### DISCUSSION TITLE
failed to run a 10x Genomics Visium Spatial as a Seurat object

### DISCUSSION THREAD

**Discussion by trandoanduyhai** [satijalab/seurat#6726] [Q&A]
We loaded output of spaceranger using Load10X_Spatial () (without error). Then we ran SpatialFeaturePlot and got this error:
Error in FUN(left, right) : non-numeric argument to binary operator

Does somebody know how to fix this problem?
Thank you very much!
Hai

**Comment by ggomes77** [comment]
The error message "non-numeric argument to binary operator" suggests that there is an issue with the data type of the gene expression values in your Seurat object. The SpatialFeaturePlot function expects numeric values for plotting, so it's possible that the gene expression values were not properly processed or loaded as numeric.

To resolve this issue, you can try the following steps:

    Make sure that you have properly loaded and processed your spatial transcriptomics data using the Load10X_Spatial function. Check that the gene expression values are correctly assigned and stored as numeric values.

    Verify the data type of the gene expression values in your Seurat object. You can use the class() function to check the data type. For example, if your Seurat object is named Brain, you can run class(Brain@assays$Spatial@data) to see the data type of the gene expression values.

    If the data type is not numeric, you may need to convert it to numeric using the as.numeric() function. You can do this by modifying your code as follows:

# Convert gene expression values to numeric
Brain@assays$Spatial@data <- as.numeric(Brain@assays$Spatial@data)

By explicitly converting the gene expression values to numeric using as.numeric(), you ensure that they are in the correct format for plotting.
